File: cgi-bin/cgi_get.pl
#!/usr/bin/perl
# FILEPATH: /Users/hlachkar/webserv/cgi-bin/cgi_get.pl
use strict;
use warnings;
use CGI;
use CGI::Carp qw(fatalsToBrowser);
use CGI::Cookie;


use CGI;
use CGI::Cookie;

my $cgi = CGI->new;
my $cookie_string = $ENV{HTTP_COOKIE};
my $first_name = $cgi->param('first_name');
my $last_name = $cgi->param('last_name');

print $cgi->header(-type=>'text/html', -charset=>'utf-8');
print "<html>";
print "<head>";
print "<title>Hello - Second CGI Program</title>";
print "</head>";
print "<body>";
print "<h2>Hello $first_name $last_name</h2>";
if ($cookie_string) {
    my $cookie = CGI::Cookie->parse($cookie_string);
    if (exists $cookie->{'lala'}) {
        my $lala_value = $cookie->{'lala'}->value;
        print "Value of cookie 'lala': $lala_value";
    } else {
        print "Cookie 'lala' does not exist";
    }
}
print "</body>";
print "</html>";

File: cgi-bin/cgi_get.py
#!/usr/bin/python3
# Import modules for CGI handling
import cgi, cgitb
import os

# BEGIN: 7d5f8a9j4d3h
import http.cookies

cookie_string = os.environ.get('HTTP_COOKIE')
# END: 7d5f8a9j4d3h
form = cgi.FieldStorage()
# Get data from fields
first_name = form.getvalue('first_name')
last_name = form.getvalue('last_name')
print ("<html>")
print ("<head>")
print ("<title>Hello - Second CGI Program</title>")
print ("</head>")
print ("<body>")
print ("<h2>Hello %s %s</h2>" % (first_name, last_name))
if cookie_string:
    cookie = http.cookies.SimpleCookie()
    cookie.load(cookie_string)
    if 'lala' in cookie:
        lala_value = cookie['lala'].value
        print(f"Value of cookie 'lala': {lala_value}")
    else:
        print("Cookie 'lala' does not exist")
print ("</body>")
print ("</html>")